scripts/etc/init.d/rcS

```diff
@@ -217,19 +232,26 @@ for i in `seq 1 10`; do
     echo -n "$i.. "
 done
 
+if [ "$ifname" != "eth0" ]; then
+    # if eth0 is not the chosen eth interface it is a wireless interface and wpa_supplicant must connect to wlan
+    echo -n "Running wpa_supplicant... "
+    wpa_supplicant -B -Dwext -c/bootfs/config/wpa_supplicant.conf -iwlan0 >& /dev/null || fail
+    [ $? -eq 0 ] && echo "OK"
```

Comment:
You can do a simple `echo "OK"` here since `|| fail` already means a return value != 0 will fail it.
Besides, if you check the return value, you should also handle the non-zero case.